### PR TITLE
ADEN-7257 Integration with AdEngine 3 bidders

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: app/modules/ads/setup.js [hunk 1]
```diff
@@ -42,8 +42,36 @@ function setupAdContext(adsContext, instantGlobals, isOptedIn = false) {
...
+		context.set('bidders.prebid.pubmatic.enabled', isGeoEnabled('wgAdDriverPubMaticBidderCountries'));
+		context.set('bidders.prebid.rubicon.enabled', isGeoEnabled('wgAdDriverRubiconPrebidCountries'));
+		context.set('bidders.prebid.rubiconDisplay.enabled', isGeoEnabled('wgAdDriverRubiconDisplayPrebidCountries'));
+
+		context.set('custom.rubiconInFV',
+			isGeoEnabled('wgAdDriverRubiconVideoInFeaturedVideoCountries') && hasFeaturedVideo);
+		context.set('custom.isCMPEnabled', isGeoEnabled('wgEnableCMPCountries'));
+	}
+
+	context.set('options.maxDelayTimeout', instantGlobals.wgAdDriverDelayTimeout || 5000);
```
Comment (nandy-andy) [inline app/modules/ads/setup.js]: What are the reasons for changing the default from `2000` to `5000`?

Comment (dmnsobczak) [reply app/modules/ads/setup.js]: No reason, my bad. Thanks for being vigilant :)

File: app/modules/ads/setup.js [hunk 2]
```diff
@@ -42,8 +42,36 @@ function setupAdContext(adsContext, instantGlobals, isOptedIn = false) {
...
+		context.set('bidders.prebid.rubicon.enabled', isGeoEnabled('wgAdDriverRubiconPrebidCountries'));
+		context.set('bidders.prebid.rubiconDisplay.enabled', isGeoEnabled('wgAdDriverRubiconDisplayPrebidCountries'));
+
+		context.set('custom.rubiconInFV',
+			isGeoEnabled('wgAdDriverRubiconVideoInFeaturedVideoCountries') && hasFeaturedVideo);
+		context.set('custom.isCMPEnabled', isGeoEnabled('wgEnableCMPCountries'));
+	}
+
 	context.set('options.maxDelayTimeout', instantGlobals.wgAdDriverDelayTimeout || 2000);
-	// TODO: context.push('delayModules', featuredVideoDelay);
```
Comment (fraszczakszymon) [inline app/modules/ads/setup.js]: Seems still valid ;)

Comment (dmnsobczak) [reply app/modules/ads/setup.js]: Right, added back :)

File: app/modules/ads/setup.js [hunk 3]
```diff
@@ -107,6 +78,46 @@ function setupAdContext(adsContext, instantGlobals, isOptedIn = false) {
...
+		context.set('bidders.prebid.openx.enabled', isGeoEnabled('wgAdDriverOpenXPrebidBidderCountries'));
+		context.set('bidders.prebid.pubmatic.enabled', isGeoEnabled('wgAdDriverPubMaticBidderCountries'));
+		context.set('bidders.prebid.rubicon.enabled', isGeoEnabled('wgAdDriverRubiconPrebidCountries'));
+		context.set('bidders.prebid.rubiconDisplay.enabled', isGeoEnabled('wgAdDriverRubiconDisplayPrebidCountries'));
+
+		const provider = context.get('state.isMobile') ? 'mobile' : 'gpt';
+		const s1 = adsContext.targeting.wikiIsTop1000 ? context.get('targeting.s1') : 'not a top1k wiki';
+
+		context.set('bidders.prebid.targeting', {
+			src: [provider],
```
Comment (fraszczakszymon) [inline app/modules/ads/setup.js]: We can hardcode 'mobile' here

Comment (dmnsobczak) [reply app/modules/ads/setup.js]: Done.

**Review by fraszczakszymon — APPROVED** [review]

**Comment by dmnsobczak** [comment]
LGTM!

**Pull request merged**